x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -1151,24 +1162,28 @@ describe('CredentialsLogic', () => {
     it('will call an API endpoint and set the results with the `onApiKeyDelete` action', async () => {
       mount();
       jest.spyOn(CredentialsLogic.actions, 'onApiKeyDelete').mockImplementationOnce(() => {});
-      (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(Promise.resolve());
+      const promise = Promise.resolve();
+      (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(promise);
 
       CredentialsLogic.actions.deleteApiKey(tokenName);
       expect(HttpLogic.values.http.delete).toHaveBeenCalledWith(
         `/api/app_search/credentials/${tokenName}`
       );
-      await flushPromises();
+      await promise;
       expect(CredentialsLogic.actions.onApiKeyDelete).toHaveBeenCalledWith(tokenName);
     });
 
     it('handles errors', async () => {
       mount();
-      (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(
-        Promise.reject('An error occured')
-      );
+      const promise = Promise.reject('An error occured');
+      (HttpLogic.values.http.delete as jest.Mock).mockReturnValue(promise);
 
       CredentialsLogic.actions.deleteApiKey(tokenName);
-      await flushPromises();
+      try {
+        await promise;
+      } catch {
+        // Ignore this
+      }
       expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
```

Comment:
If you wrap the expect() inside the catch you skip having to have `// Ignore this` - looks a little cleaner and matches the syntax in the source code! :)
```suggestion
      } catch {
        expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
      }
```

Reply:
Ah, true. The thing that makes me a little nervous (generally) about code like that is if you somehow get yourself in a situation where the code *doesn't* reject the promise, and you get a false positive on this test since your expect is not executed.

But in this case, I think it's OK, because we're explicitly rejecting our promise IN the test code, so it should always trigger.

Reply:
For sure - we definitely only want this try/catch pattern in tests where we're expecting an error :)